zkvm/spec/ZkVM.md

```diff
@@ -1278,14 +1290,19 @@ _ex1 ex2_ **mul** → _ex3_
 
 Multiplies two [expressions](#expression-type) producing another [expression](#expression-type) representing the result of multiplication.
 
-This performs an optimization: if one of the expressions `ex1` or `ex2` contains only one term and this term is for the variable representing the R1CS constant `1` (in other words, the statement is a cleartext constant), then the other expression is multiplied by that constant in-place without allocating a multiplier in the [constraint system](#constraint-system).
+This performs a _guaranteed optimization_: if one of the expressions `ex1` or `ex2` contains
```

Comment:
nit: Could we remove the explanation for constant expression?

It would then read as: "This performs a guaranteed optimization if one of the expressions, ex1 or ex2 is a cleartext constant. The other expression is multiplied by that constant in-place without allocating a multiplier in the constraint system."

WDYT?

Reply:
cleartext constant != constant expression. The first is a Scalar, the second is Expression. The idea was to not have a separate instruction for mul-by-scalar distinct from mul-by-expression.